Duplicati/Library/Main/Operation/BackupHandler.cs

```diff
@@ -271,15 +271,17 @@ private void PostBackupVerification()
                 backend.WaitForComplete(m_database, null);
             }
 
-            if (m_options.BackupTestSampleCount > 0 && m_database.GetRemoteVolumes().Any())
+            long remoteVolumeCount = m_database.GetRemoteVolumes().LongCount(x => x.State == RemoteVolumeState.Verified);
+            long samplesToTest = Math.Max(m_options.BackupTestSampleCount, (long)Math.Round(remoteVolumeCount * (m_options.BackupTestPercentage / 100D), MidpointRounding.AwayFromZero));

```

Comment:
Why do you multiple with the number of remote volumes?

Reply:
I believe this should yield the number of files to test, which is what needs to be provided to `TestHandler.DoRun`.  The `GetRemoteVolumes` method returns entries for all the `dindex`, `dlist`, and `dblock` files on the backend.

Reply:
I'm not familiar with this part of the code yet, but I understood from the issue you referred to that this percentage is with respect to the number of files in a backup set that should be tested. What is the meaning of a remote volume ? If it corresponds to a backed up file on a remote backend, then I understand.

Reply:
My understanding is that both the `backup-test-samples` and `backup-test-percentage` options refer to the files on the backend.  For example, if the number of samples to test is 3, then it will attempt to verify up to 3 `dindex` files, 3 `dlist` files, and 3 `dblock` files, for a total of up to 9 files.  The total number of files tested is reflected in `TestResults: VerificationsActualLength` in the backup log.

The method that selects the files to be tested is [here](https://github.com/duplicati/duplicati/blob/master/Duplicati/Library/Main/Database/LocalTestDatabase.cs#L112).

Reply:
Ok, but how many actual files from the backup set are contained in dindex? If there's one dblock containing the entire backup set, and I specify 1% of my files should be verified, then nothing gets verified due to the rounding. This is a border case but still. So this percentage is with respect to blocks. Is it possible to find out, how many files are in a block and guarantee at least this percentage of files is checked? I'm also not sure what the best solution is, because I guess you can only verify blockwise.

Reply:
The example you mention is the reason why we take the maximum of the samples implied by the `backup-test-samples` and `backup-test-percentage` options.  If there's a better approach, please let me know.

I don't think it's possible to guarantee how many user files are verified, since a file's blocks may be spread out across many `dblock` files.

Reply:
I like the balance between always validating at least 3, but up to a percentage. 

I don't think worrying about individual user files makes sense. It's bad for performance, bad for user understanding, and it doesn't actually matter for validation what files you validate as long as you validate all volumes eventually.

Reply:
> The example you mention is the reason why we take the maximum of the samples implied by the `backup-test-samples` and `backup-test-percentage` options. If there's a better approach, please let me know.
> 
> I don't think it's possible to guarantee how many user files are verified, since a file's blocks may be spread out across many `dblock` files.

That's true. But can you then refer to blocks  instead of files in the outputs of the help command, to make this more specific?

Reply:
One difficulty is that the option actually does pertain to files (`dindex`, `dlist`, and `dblock` files), and not blocks.  I tried to be consistent with the help for the `backup-test-samples` and `full-remote-verification` options so that users would recognize that the options are related.  

If the help text is unclear, perhaps we can reword these from
> After a backup is completed, some files are selected for verification on the remote backend.

to

> After a backup is completed, some files from the remote backend are selected for verification.

If that makes sense, I'd prefer that we make the change in a separate pull request as it feels a bit out of scope here.  The text in the user's manual should then be changed as well to be consistent.

Reply:
Seems fair

Reply:
I agree that in the already existing option it is also not precise with the phrasing 'some files' and I think this should be fixed. I know that they are indeed files, but we can say precisely what kind of files these are: the dlist, dblock and dindex files. One could mistake them for user files, which I would have had first in mind with a backup program and without in depth knowledge of the internals of it. And this is not true.

I hope you see my point with that :) I will merge it in and hope you or otherwise I will create a merge request with the help.

@Pectojin : I agree with that.

Reply:
That makes sense to me.  I'll create a pull request with the above mentioned modification to the help and we can continue the discussion there.

Reply:
Pull request #3585 was created to clarify the help text.